common/web/keyboard-processor/src/keyboards/keyboardLoaderBase.ts

```diff
@@ -19,5 +20,48 @@ export default abstract class KeyboardLoaderBase {
     return promise;
   }
 
-  protected abstract loadKeyboardInternal(uri: string): Promise<Keyboard>;
+  public loadKeyboardFromStub(stub: KeyboardProperties & { filename: string }) {
+    this.harness.install();
+    let promise = this.loadKeyboardInternal(stub.filename, stub.id);
+
+    promise = promise.catch((err: Error) => {
+      if(err == KeyboardLoaderBase.scriptErrorMessage(stub.filename)) {
+        // Enhance the error message.
+        err.message = KeyboardLoaderBase.scriptErrorMessage(stub);
+      } else if(err == KeyboardLoaderBase.missingErrorMessage(stub.filename)) {
+        // Same thing here.
+        err.message = KeyboardLoaderBase.scriptErrorMessage(stub);
+      }
```

Comment:
This is all a little magical. I don't really understand what you are doing here.

Also are you meaning line 33 to have `scriptErrorMessage` or `missingErrorMessage`?

Reply:
Sounds like I should probably retool things a little bit to make the error-handling easier to follow, then.  I did feel that way when I did these changes, and figured I could triage to later... but if it's too difficult to follow in the PR that makes the change, then it probably should be made clearer here, now.

If you want details - the method that actually does the loading operates off of just the filename, without knowledge of the keyboard name and language name associated with the load request.  As a result, that method can only give the `stub.filename`-based version of the error.  _This_ method - the one loading from a stub - uses that, and so maps the filename-only error to the variant with richer feedback data.

That said, it may be best to retool things a bit so that there's no need to 'error map' in the first place.